### PR TITLE
Bugfix on merging of non-bcsc records with bcsc records.

### PR DESCRIPTION
**Pull request by amit-metri**
-bugfix on merging of non-bcsc records with bcsc records.

### CODE REVIEW

**Review by pinakinkansara — APPROVED** [review]

File: data/src/main/java/ca/bc/gov/data/datasource/local/dao/TestResultDao.kt [hunk 1]
```diff
@@ -32,6 +32,6 @@ interface TestResultDao {
     @Query("SELECT id FROM test_result WHERE patient_id = :patientId AND collection_date = :collectionDate")
     suspend fun getTestResultId(patientId: Long, collectionDate: Instant): Long?
 
-    @Query("DELETE FROM test_result WHERE patient_id = :patientId AND data_source = 'BCSC'")
+    @Query("DELETE FROM test_result WHERE patient_id = :patientId")
     suspend fun deleteAuthenticatedTestRecords(patientId: Long): Int
```
Comment (pinakinkansara) [inline data/src/main/java/ca/bc/gov/data/datasource/local/dao/TestResultDao.kt:36]: rename method to delete()

File: data/src/main/java/ca/bc/gov/data/datasource/local/dao/MedicationRecordDao.kt [hunk 2]
```diff
@@ -20,6 +20,6 @@ interface MedicationRecordDao : BaseDao<MedicationRecordEntity> {
     @Query("SELECT * FROM medication_record WHERE id = :medicationRecordId")
     suspend fun getMedicationWithSummaryAndPharmacy(medicationRecordId: Long): MedicationWithSummaryAndPharmacy?
 
-    @Query("DELETE FROM medication_record WHERE patient_id = :patientId AND data_source = 'BCSC'")
+    @Query("DELETE FROM medication_record WHERE patient_id = :patientId")
     suspend fun deleteAuthenticatedMedicationRecords(patientId: Long): Int
```
Comment (pinakinkansara) [inline data/src/main/java/ca/bc/gov/data/datasource/local/dao/MedicationRecordDao.kt:24]: rename method to delete()

File: data/src/main/java/ca/bc/gov/data/datasource/local/dao/VaccineRecordDao.kt [hunk 3]
```diff
@@ -44,6 +44,6 @@ interface VaccineRecordDao {
     @Query("SELECT * FROM vaccine_dose WHERE vaccine_record_id = :vaccineRecordId")
     suspend fun getVaccineDoses(vaccineRecordId: Long): List<VaccineDoseEntity>
 
-    @Query("DELETE FROM vaccine_record WHERE patient_id = :patientId AND data_source = 'BCSC'")
+    @Query("DELETE FROM vaccine_record WHERE patient_id = :patientId")
     suspend fun deleteAllAuthenticatedVaccineRecord(patientId: Long): Int
```
Comment (pinakinkansara) [inline data/src/main/java/ca/bc/gov/data/datasource/local/dao/VaccineRecordDao.kt:48]: rename method to delete()

**Review by pinakinkansara — CHANGES_REQUESTED** [review]

File: data/src/main/java/ca/bc/gov/data/datasource/local/dao/TestResultDao.kt (hunk 1 above)
Comment (amit-metri) [reply data/src/main/java/ca/bc/gov/data/datasource/local/dao/TestResultDao.kt:36]: renamed to deleteTestRecords() as delete() already exists

File: data/src/main/java/ca/bc/gov/data/datasource/local/dao/VaccineRecordDao.kt (hunk 3 above)
Comment (amit-metri) [reply data/src/main/java/ca/bc/gov/data/datasource/local/dao/VaccineRecordDao.kt:48]: renamed to deletePatientVaccineRecords() as delete already exists

File: data/src/main/java/ca/bc/gov/data/datasource/local/dao/MedicationRecordDao.kt (hunk 2 above)
Comment (amit-metri) [reply data/src/main/java/ca/bc/gov/data/datasource/local/dao/MedicationRecordDao.kt:24]: renamed to deletePatientTestRecords() as delete already exists.

**Review by pinakinkansara — APPROVED** [review]

**Pull request merged**